File: tests/fixtures/vulnerable/no_security_logging.py
"""Vulnerable: auth handlers with bare logger.info only."""
import logging

logger = logging.getLogger(__name__)


def login(request):
    user = authenticate(request.json["username"], request.json["password"])
    if not user:
        logger.info("login failed")
        return {"error": "Invalid credentials"}, 401
    return {"token": create_token(user)}, 200

File: tests/fixtures/vulnerable/no_security_logging.ts
// Vulnerable: auth handler with only console.log
export async function login(req, res) {
  const user = await authenticate(req.body.username, req.body.password);
  if (!user) {
    console.log("login failed");
    return res.status(401).json({ error: "Invalid credentials" });
  }
  return res.json({ token: createToken(user.id) });
}
